mailpoet/assets/js/src/date.ts

```diff
@@ -45,15 +45,15 @@ export const MailPoetDate = {
     let momentDate;
     this.init(options);
 
-    momentDate = Moment(date, this.convertFormat(options.parseFormat));
+    momentDate = Moment(date);
     if (options.offset === 0) momentDate = momentDate.utc();
     return momentDate.format(this.convertFormat(this.options.format));
   },
   toDate: function toDate(date: MomentInput, opts?: DateOptions): Date {
     const options = opts || {};
     this.init(options);
 
-    return Moment(date, this.convertFormat(options.parseFormat)).toDate();
+    return Moment(date).toDate();
```

Comment:
What does this do? Seems like it could change the behavior somehow...

Reply:
Thanks! Good catch.

The original message was: `Unsafe argument of type `any` assigned to a parameter of type `boolean`  @typescript-eslint/no-unsafe-argument` which I attributed to the Moment declaration, where I missed the overload:
![image](https://user-images.githubusercontent.com/6458412/154923381-837d66e6-26ab-495b-a0b7-f4d9020351b7.png)

Instead it is related to `this.convertFormat()`returning _allegedly_ `any`, while it actually defines `string` as it's return value. I tried several things to explain this to TS but failed, that's why 04a935d66 ends up typecasting the output in the Moment constructor. Probably I am missing here something, why TS does not as expected:
![image](https://user-images.githubusercontent.com/6458412/154924346-e5e6b27c-3396-4bef-ab99-f61c5e36b755.png)
![image](https://user-images.githubusercontent.com/6458412/154924437-b0e73a93-6cf8-4b0c-bdd8-e7aa5032cb63.png)

Reply:
@websupporter Is this maybe because `this.convertFormat` is a property on the Moment object? It's defined as:
```ts
export const MailPoetDate = {
  ...
  convertFormat: function convertFormat(format: string): string { ... }
  ...
}
```

In that case does it help to add `as const` to the object? Like:
```ts
export const MailPoetDate = {
  ...
  convertFormat: function convertFormat(format: string): string { ... }
  ...
} as const;
```